cyvcf2/cyvcf2.pyx

```diff
@@ -670,6 +670,41 @@ cdef class VCF(HTSFile):
             stdlib.free(cnames)
             return self._seqnames
 
+    cdef _num_records(self):
+        cdef uint64_t total, records, v;
+        cdef int ret, tid, nseq;
+        cdef hts_idx_t *idx = NULL;
+
+        if self.hidx == NULL and self.idx == NULL:
```

Comment:
not required for this PR, but perhaps we should extract out this stuff to a `_find_index` method since it's repeated elsewhere.

Reply:
I've created an ``_open_index`` function that's shared between num_records and seqnames, as they are very similar. I'm a bit wary of the other places the indexes are used, as it's really not clear to me when you have to use the tbx struct (which has an embedded hts_idx) and when you can use the other for queries. I thought it best to leave well enough alone for now.